scripts/deployment/extract_pools.sh

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env bash
+set -e
+
+# Import the deploy_liquidity_hub script
+deployment_script_dir=$(cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)
+project_root_path=$(realpath "$0" | sed 's|\(.*\)/.*|\1|' | cd ../ | pwd)
+
+# Displays tool usage
+function display_usage() {
+  echo "WW Pool Deployer"
+  echo -e "\nUsage:./deploy_pool.sh [flags].\n"
+  echo -e "Available flags:\n"
+  echo -e "  -h \thelp"
+  echo -e "  -c \tThe chain where you want to deploy (juno|juno-testnet|terra|terra-testnet|... check chain_env.sh for the complete list of supported chains)"
+  echo -e "  -p \tPool configuration file to get deployment info from."
+}
+
+function generate_label() {
+  local assets=("$1" "$2")
+  local processed_assets=()
+
+  for asset in "${assets[@]}"; do
+    local processed_asset="$asset"
+    if [[ "$asset" == ibc/* ]]; then
+      denom_trace_result=$($BINARY q ibc-transfer denom-trace $asset --node $RPC -o json | jq -r '.denom_trace.base_denom')
+
+      if [[ "$denom_trace_result" == factory/* ]]; then
+        processed_asset=$(echo "$denom_trace_result" | awk -F'/' '{print $NF}')
+      elif [[ "$denom_trace_result" == factory:* ]]; then
+        processed_asset="${denom_trace_result##*:}"
+      else
+        processed_asset="$denom_trace_result"
+      fi
+
+    elif [[ "$asset" == factory/* ]]; then
+      processed_asset=$(basename "$asset")
+    elif [[ "$asset" == $chain* ]]; then
+      local query='{"token_info":{}}'
+      local symbol=$($BINARY query wasm contract-state smart $asset "$query" --node $RPC --output json | jq -r '.data.symbol')
+      processed_asset=$symbol
+    fi
+    processed_assets+=("$processed_asset")
+  done
```

Comment:
Pls comment this. Kinda hard to follow with all the if else branching.

Reply:
Sure, will do